core/azure-core/src/main/java/com/azure/core/polling/PollResponse.java

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.polling;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Container of information related to polling and poll response. It will hold current polling status {@link OperationStatus} , T value,
+ * retry after duration and various configuration properties.
+ *
+ * <p><strong>Container of PollResponse and related information.</strong></p>
+ * @param <T> type of poll response value
+ */
+public final class PollResponse<T> {
+
+    private OperationStatus status;
+    private T value;
+    private Duration retryAfter;
+    private Map<Object, Object> properties;
+
+    /**
+     * Represent various state of poll operation.
+     * The poll operation is considered complete/done when status is one of SUCCESSFULLY_COMPLETED/FAILED/USER_CANCELLED.
+     **/
+    public enum OperationStatus {
+        NOT_STARTED,
+        IN_PROGRESS,
+        SUCCESSFULLY_COMPLETED,
+        FAILED,
+        USER_CANCELLED
+    }
+
+    /**
+     * Create a Poll Response
```

Comment:
Be more verbose everywhere. "Creates a new PollResponse instance to represent the current state of a single polling operation", for example.

Reply:
resolved